cloudflare/data_source_waf_groups.go

```diff
@@ -0,0 +1,179 @@
+package cloudflare
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func dataSourceCloudflareWAFGroups() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceCloudflareWAFGroupsRead,
+
+		Schema: map[string]*schema.Schema{
+			"zone_id": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
This field is required, yet not documented.

Reply:
Fixed! As well as for the `package_id` field.